### ISSUE TITLE
Add directory creation method to FTPClient

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3500]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3500|
|Type|enhancement|
|Created|2008-10-24 22:30:57Z|
|Branch|https://github.com/twisted/twisted/tree/ftp-mkdir-3500|

FTP supports directory creation.  Add a handy helper to `FTPClient` to access this.

<details><summary>Searchable metadata</summary>

```
trac-id__3500 3500
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_ftp_mkdir_3500 branches/ftp-mkdir-3500
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ftp ftp
keywords__ 
time__1224887457000000 1224887457000000
changetime__1225375442000000 1225375442000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [25085]) Branching to 'ftp-mkdir-3500'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [25086]) apply the makeDirectory parts of ftp_rename_patch_v5.diff

refs [#3335](https://github.com/twisted/twisted/issues/3335)
refs [#3500](https://github.com/twisted/twisted/issues/3500)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

alright, should be cool

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

an ```@since``` tag is missing and client.transport.loseConnection() should have epytext markup.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [25103]) improve docstring markup; revert unrelated change to quit docstring

refs [#3500](https://github.com/twisted/twisted/issues/3500)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_7"></a>|@radix commented|
|-|-|

Please change the @since tag to say "Twisted 8.2" instead of just "8.2".

After that you can merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @radix|
|-|-|

That conflicts with the coding standard.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_9"></a>|@radix set owner to @exarkun|
|-|-|

OK, I guess we'll wait until after this release to update the coding standard. Go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

(In [25187]) fix intermittent crummyness

refs [#3500](https://github.com/twisted/twisted/issues/3500)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set owner to @radix|
|-|-|

Oops, there was a bug.  Please take another look.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_12"></a>|@radix set owner to @exarkun|
|-|-|

ok that's fine please merge

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [25194]) Merge ftp-mkdir-3500

Author: tarjei, exarkun
Reviewer: radix
Fixes: [#3500](https://github.com/twisted/twisted/issues/3500)

Add `FTPClient.makeDirectory` method for using the directory creation feature of FTP.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
